### ISSUE TITLE
Library does not build 

### ISSUE THREAD

**Issue by OlivierDehaene** [launchdarkly/rust-server-sdk#13]
**Describe the bug**
The library does not build because of a change made in https://github.com/launchdarkly/rust-server-sdk-evaluation

**To reproduce**
```shell
git clone git@github.com:launchdarkly/rust-server-sdk.git
cd rust-server-sdk
cargo build
```

or 

```shell
mkdir example-app
cd example-app
cargo init 
echo 'launchdarkly-server-sdk = "1.0.0-beta.2"' >> Cargo.toml
cargo build
```

**Logs**
```shell
error[E0308]: mismatched types
   --> launchdarkly-server-sdk/src/evaluation.rs:169:32
    |
169 |                     variation: detail.variation_index,
    |                                ^^^^^^^^^^^^^^^^^^^^^^ expected `usize`, found `isize`
    |
    = note: expected enum `std::option::Option<usize>`
               found enum `std::option::Option<isize>`

For more information about this error, try `rustc --explain E0308`.
error: could not compile `launchdarkly-server-sdk` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed
```

**SDK version**
`1.0.0-beta.2`

**Comment by keelerm84** [comment]
Thank you for raising this issue. We will work on getting that addressed. 

Filed internally as SC-148545

**Comment by keelerm84** [comment]
This should be resolved with the latest release (1.0.0-beta.3)
